cmd/forwarder/http.go

```diff
@@ -87,76 +97,88 @@ func extractRemoteAddr(r *http.Request) string {
 	return remoteAddr
 }
 
-func (s *httpServer) Run() error {
-	go s.awaitShutdown()
+func (s *httpServer) handleLogs(w http.ResponseWriter, r *http.Request) {
+	defer s.posts.UpdateSince(time.Now())
 
-	//FXME: check outlet depth?
-	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
-		defer s.healthChecks.UpdateSince(time.Now())
-		if s.isShuttingDown {
-			http.Error(w, "Shutting down", 503)
-			return
-		}
+	if s.Config.EnforceSsl && r.Header.Get("X-Forwarded-Proto") != "https" {
+		s.handleHTTPError(w, "Only SSL requests accepted", 400)
+		return
+	}
+
+	if s.isShuttingDown {
+		s.handleHTTPError(w, "Shutting down", 503)
+		return
+	}
 
-	})
+	if r.Method != "POST" {
+		s.handleHTTPError(w, "Only POST is accepted", 400)
+		return
+	}
 
-	http.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
-		defer s.posts.UpdateSince(time.Now())
+	contentType := r.Header.Get("Content-Type")
+	if !s.validContentType(contentType) {
+		s.handleHTTPError(w, "Only Content-Type application/logplex-1 is accepted", 400)
+		return
+	}
 
-		if s.Config.EnforceSsl && r.Header.Get("X-Forwarded-Proto") != "https" {
-			s.handleHTTPError(w, "Only SSL requests accepted", 400)
-			return
-		}
+	if !s.auth.Authenticate(r) {
+		s.pAuthErrors.Inc(1)
+		s.handleHTTPError(w, "Unable to authenticate request", 401)
+		return
+	}
 
-		if s.isShuttingDown {
-			s.handleHTTPError(w, "Shutting down", 503)
-			return
-		}
+	s.pAuthSuccesses.Inc(1)
 
-		if r.Method != "POST" {
-			s.handleHTTPError(w, "Only POST is accepted", 400)
-			return
-		}
+	remoteAddr := extractRemoteAddr(r)
+	requestID := r.Header.Get("X-Request-Id")
+	logplexDrainToken := r.Header.Get("Logplex-Drain-Token")
```

Comment:
Perhaps this is too pendantic, but consider `s/logplexDrainToken/drainToken/g`, given that it's for both LogPlex and Msgpack?

Reply:
Can do